src/metabase/pulse.clj

```diff
@@ -541,6 +543,8 @@
 (defmethod send-notification! :slack
   [{:keys [channel-id message attachments]}]
   (let [attachments (create-and-upload-slack-attachments! attachments)]
+    ;; Cal 2024-03-04: Without this sleep, attached images don't always appear. I tried with 1000ms and it wasn't enough.
+    (Thread/sleep 2000)
```

Comment:
Frankly I don't know why this is happening. I can only guess it's an issue with slack's state.

Without this sleep, instead of getting a result like this:
<img width="396" alt="image" src="https://github.com/metabase/metabase/assets/39073188/df45e6f7-e8c7-4451-9f6e-a2299adf314b">

I would get this:
<img width="284" alt="image" src="https://github.com/metabase/metabase/assets/39073188/55bbda81-48b6-40bf-b2e4-d2a52322f069">

even though the sent message and attachments were the same in both cases, based on the logs
```
2024-04-29 17:06:54,493 TRACE integrations.slack :: Slack API request: "https://slack.com/api/chat.postMessage" {:form-params {:channel "@cal", :username "MetaBot", :icon_url "http://static.metabase.com/metabot_slack_avatar_whitebg.png", :text nil, :attachments "[{\"blocks\":[{\"type\":\"header\",\"text\":{\"type\":\"plain_text\",\"text\":\"subdash\",\"emoji\":true}},{\"type\":\"section\",\"fields\":[{\"type\":\"mrkdwn\",\"text\":\"Sent by Crowberto Corv\"}]}]},{\"title\":\"Orders, Count\",\"title_link\":\"http://localhost:3001/question/30\",\"fallback\":\"Orders, Count\",\"text\":\"18,760\"},{\"title\":\"Best selling products\",\"title_link\":\"http://localhost:3001/question/23\",\"fallback\":\"Best selling products\",\"image_url\":\"https://files.slack.com/files-pri/T078VCLCR-F071306S06R/image.png\"},{\"blocks\":[{\"type\":\"divider\"},{\"type\":\"context\",\"elements\":[{\"type\":\"mrkdwn\",\"text\":\"<http://localhost:3001/dashboard/2|*Sent from Metabase Test*>\"}]}]}]"}}
```

Reply:
I hit this again with a 2000ms sleep, so I've extended it to 3000ms. 

The issue appears to be caused by a race condition between the message being sent to the subscription's destination, and the actual upload of the file to the files channel. The latter needs to finish first, and the former needs to finish afterwards. The race condition is happening on the slack side, because on our side we're calling 

Here's when the app integration uploaded the image to the files channel, from slacks' point of view:
<img width="411" alt="image" src="https://github.com/metabase/metabase/assets/39073188/5ed69117-966a-4c16-938a-58c0a4d7b86f">

And here's when the message was sent to me, from slack's point of view:
<img width="261" alt="image" src="https://github.com/metabase/metabase/assets/39073188/9862eea5-0ac4-47b7-99b4-f6a6d22ef7e3">

The upload request ([files.completeUploadExternal](https://api.slack.com/methods/files.completeUploadExternal)) was completed first, but in slack they appear the other way around.

Reply:
Seems like others are running into this issue too. https://github.com/slackapi/python-slack-sdk/issues/1363

We could try polling `[conversations_history](https://api.slack.com/methods/conversations.history)` on the file channel, but that seems pretty invasive and I wonder if we're allowed to do that.

> using conversations_history is one of the recommended approaches to retrieve your message's details such as the timestamp

Reply:
Hard to believe slack screwed up this API so bad. They must have been in a rush